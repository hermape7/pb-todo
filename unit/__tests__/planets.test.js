const {
  getPlanetByName,
  getAllPlanets,
  getPlanet
} = require("../js/sw-client");

describe("Planet Tests", () => {
  describe("GET /api/planets", () => {
    const planetNameToBeFind = "Tatooine";
    const planetNameToNotBeFind = "definitely-not-in-the-list";

    it("GET all planets", async () => {
      const allPlanets = await getAllPlanets();
      expect(allPlanets.length).toBe(60);
    });

    it(`GET planet ${planetNameToBeFind} should be in the list`, async () => {
      const planet = await getPlanetByName(planetNameToBeFind);
      expect(planet.name).toBe(planetNameToBeFind);
      expect(planet).toMatchSnapshot({
        created: expect.any(String),
        edited: expect.any(String)
      });
    });

    it(`GET planet ${planetNameToNotBeFind} should not be in the list`, async () => {
      const planet = await getPlanetByName("not-in-the-list");
      expect(planet).toBeFalsy();
    });
  });

  describe("GET /api/planets/:id", () => {
    const planetToBeFind = {
      id: 21,
      name: "Eriadu"
    };

    it(`GET planet with id ${planetToBeFind.id} should be in the list`, async () => {
      const planet = await getPlanet(planetToBeFind.id);
      expect(planet).toMatchSnapshot({
        created: expect.any(String),
        edited: expect.any(String)
      });
    });

    it("GET planet with id that is not in the list", async () => {
      const planet = await getPlanet(324547893);
      expect(planet).toHaveProperty("detail", "Not found");
    });
  });
});
