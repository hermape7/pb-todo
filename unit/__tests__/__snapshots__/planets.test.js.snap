// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Tests GET /api/planets GET planet Tatooine should be in the list 1`] = `
Object {
  "climate": "arid",
  "created": Any<String>,
  "diameter": "10465",
  "edited": Any<String>,
  "films": Array [
    "http://swapi.dev/api/films/1/",
    "http://swapi.dev/api/films/3/",
    "http://swapi.dev/api/films/4/",
    "http://swapi.dev/api/films/5/",
    "http://swapi.dev/api/films/6/",
  ],
  "gravity": "1 standard",
  "name": "Tatooine",
  "orbital_period": "304",
  "population": "200000",
  "residents": Array [
    "http://swapi.dev/api/people/1/",
    "http://swapi.dev/api/people/2/",
    "http://swapi.dev/api/people/4/",
    "http://swapi.dev/api/people/6/",
    "http://swapi.dev/api/people/7/",
    "http://swapi.dev/api/people/8/",
    "http://swapi.dev/api/people/9/",
    "http://swapi.dev/api/people/11/",
    "http://swapi.dev/api/people/43/",
    "http://swapi.dev/api/people/62/",
  ],
  "rotation_period": "23",
  "surface_water": "1",
  "terrain": "desert",
  "url": "http://swapi.dev/api/planets/1/",
}
`;

exports[`Planet Tests GET /api/planets/:id GET planet with id 21 should be in the list 1`] = `
Object {
  "climate": "polluted",
  "created": Any<String>,
  "diameter": "13490",
  "edited": Any<String>,
  "films": Array [],
  "gravity": "1 standard",
  "name": "Eriadu",
  "orbital_period": "360",
  "population": "22000000000",
  "residents": Array [
    "http://swapi.dev/api/people/12/",
  ],
  "rotation_period": "24",
  "surface_water": "unknown",
  "terrain": "cityscape",
  "url": "http://swapi.dev/api/planets/21/",
}
`;
